### PR TITLE
History and playlist feature

### PR DESCRIPTION
**Pull request by abhisheksahani-nova**
Note :-  Ignore Backend files , thanks !

### CODE REVIEW

File: src/components/cards/LargeHorizontalCard/LargeHorizontalCard.jsx [hunk 1]
```diff
@@ -1,12 +1,25 @@
 import React from "react";
 import "./largeHorizontalCard.css";
-import {useLikedVideos} from "../../../context/likedVideos-context";
+import { useLikedVideos } from "../../../context/likedVideos-context";
```
Comment (sainath7878) [inline src/components/cards/LargeHorizontalCard/LargeHorizontalCard.jsx:3]: suggestion:- Try using barrel export. This way the code will look clean.

File: src/components/cards/LargeHorizontalCard/LargeHorizontalCard.jsx [hunk 2]
```diff
@@ -18,10 +31,32 @@ function LargeHorizontalCard({ video }) {
             alt="plant image"
           />
 
+          {location.pathname == "/historyVideos" ? (
```
Comment (sainath7878) [inline src/components/cards/LargeHorizontalCard/LargeHorizontalCard.jsx:34]: suggestion: use if else statements if there are nesting conditions.

File: src/components/cards/verticalCard/VerticalCard.jsx [hunk 3]
```diff
@@ -23,22 +28,43 @@ function VerticalCard({ video }) {
...
+          ></i>
+
+          {showPlaylistDropdown ? (
+            <PlaylistDropdown
+              setShowPlaylistDropdown={setShowPlaylistDropdown}
+              video={video}
+              hideSaveToPlaylist={false}
+            />
+          ) : (
+            ""
```
Comment (nayyyhaa) [inline src/components/cards/verticalCard/VerticalCard.jsx:56]: you can use `&&` for showing element if it has no else condition.

File: src/components/cards/verticalCard/VerticalCard.jsx [hunk 4]
```diff
@@ -23,22 +28,43 @@ function VerticalCard({ video }) {
...
               onClick={() => handleAddToWatchLater(video, token)}
             ></i>
           )}
 
+          <i
+            onClick={() => setShowPlaylistDropdown((prev) => !prev)}
+            className="fa-solid fa-file-circle-plus position-abs verticalCard-playlist-icon"
+          ></i>
+
+          {showPlaylistDropdown ? (
```
Comment (sainath7878) [inline src/components/cards/verticalCard/VerticalCard.jsx:49]: Suggestion: Here you could you conditional rendering like ``` {showPlaylistDropdown && .... }```

**Review by nayyyhaa — APPROVED** [review]
not much to suggest. code is easily understandable & well-written with proper folder-structure & variable naming. you can merge the PR.

**Review by sainath7878 — APPROVED** [review]
Everything else looks good apart from suggested changes. Good work!

File: src/components/cards/verticalCard/VerticalCard.jsx (hunk 3 above)
Comment (abhisheksahani-nova) [reply src/components/cards/verticalCard/VerticalCard.jsx:56]: Yess I will take these suggestion in next pr

File: src/components/cards/verticalCard/VerticalCard.jsx (hunk 4 above)
Comment (abhisheksahani-nova) [reply src/components/cards/verticalCard/VerticalCard.jsx:49]: Yess

File: src/components/cards/LargeHorizontalCard/LargeHorizontalCard.jsx (hunk 2 above)
Comment (abhisheksahani-nova) [reply src/components/cards/LargeHorizontalCard/LargeHorizontalCard.jsx:34]: Its if else only with ternary operator , its easier to use in jsx

File: src/components/cards/LargeHorizontalCard/LargeHorizontalCard.jsx (hunk 1 above)
Comment (abhisheksahani-nova) [reply src/components/cards/LargeHorizontalCard/LargeHorizontalCard.jsx:3]: Will look into it , whats barrel import ?

**Pull request merged**